src/controllers/AdminController.php

```diff
@@ -3255,40 +3326,55 @@ class={$highlighted_color}
           $log_entry =
             <span class="highlighted--red">{$gamelog->getEntry()}</span>;
         }
-        $team = await MultiTeam::genTeam($gamelog->getTeamId());
-        $level = await Level::gen($gamelog->getLevelId());
-        $country = await Country::gen($level->getEntityId());
-        $level_str =
-          $country->getName().
-          ' - '.
-          $level->getTitle().
-          ' - '.
-          $level->getType();
-        $logs_tbody->appendChild(
-          <tr>
-            <td>{time_ago($gamelog->getTs())}</td>
-            <td>{$log_entry}</td>
-            <td>{$level_str}</td>
-            <td>{strval($gamelog->getPoints())}</td>
-            <td>{$team->getName()}</td>
-            <td>{$gamelog->getFlag()}</td>
-          </tr>
-        );
-      }
-      $logs_table =
-        <table>
-          <thead>
+
+        $awaitables = Map {
+          'team' => MultiTeam::genTeam($gamelog->getTeamId()),
+          'level' => Level::gen($gamelog->getLevelId()),
+        };
+
+        $results = await \HH\Asio\m($awaitables);
+
+        if ($results->contains('team') && $results->contains('level')) {
+          $team = $results->get('team');
+          $level = $results->get('level');
+          /* HH_IGNORE_ERROR[4064]: Object can never be null due to line 3321 */
```

Comment:
Instead of using HH_IGNORE_ERROR here (and any other place where this applies), it's better to convince Hack that something can't be null. You can do something like `invariant($level !== null, "Level should not be null")`.

If you ever want to assert something is of the right type, you can use `is_array` for arrays or `instanceof` for objects.